apps/lib/apps.c

```diff
@@ -607,9 +608,13 @@ EVP_PKEY *load_keyparams(const char *uri, int maybe_stdin, const char *desc)
 
     (void)load_key_certs_crls(uri, maybe_stdin, NULL, desc,
                               NULL, NULL, &params, NULL, NULL, NULL, NULL);
-    if (params == NULL) {
-        BIO_printf(bio_err, "Unable to load %s\n", desc);
+    if (params != NULL && keytype != NULL && !EVP_PKEY_is_a(params, keytype)) {
+        BIO_printf(bio_err,
+                   "Unable to load %s from %s (unexpected parameters type)\n",
+                   desc, uri);
         ERR_print_errors(bio_err);
+        EVP_PKEY_free(params);
+        params = NULL;
     }
```

Comment:
You lost the error printout for `params == NULL`

(I'm not sure why we still have those error printouts, since `load_key_certs_crls()` does that already, but removing it only here makes this function inconsistent with `load_key` and `load_pubkey`)

Reply:
This change was deliberate. As per my introductory comments:

"We also fix a number of follow on issues which resulted in multiple similar errors messages being displayed for the same problem,"

>  but removing it only here makes this function inconsistent with load_key and load_pubkey

That may be true, but I consider those functions out of scope of this change.

Reply:
So, will that be fixed at some point?

Reply:
I'll do it in a separate PR. I don't want to hold up merging this one.

Reply:
#13355